src/create-subscription-orders.js

```diff
@@ -0,0 +1,243 @@
+        stats.recoverableErrorUrlCalls++
+      } else {
+        logger.error(
+          `Template order ${orderNumber}: ` +
+            `Unrecoverable error received when calling ${subscriptionOrderCreationUrl}. ` +
+            'Please check the order and set its state back to "Active" ' +
+            `Response status code ${response.status}`
+        )
+        await updateOrderWithRetry(
+          apiRoot,
+          id,
+          [
+            {
+              action: 'transitionState',
+              state: {
+                typeId: 'state',
+                key: ERROR_STATE,
+              },
+            },
+          ],
+          version
+        )
+        stats.unrecoverableErrorUrlCalls++
+      }
+    else {
+      await _updateTemplateOrder(id, version, customFieldsRaw, stateKey)
+      stats.successUrlCalls++
+    }
+  }
+  stats.processedTemplateOrders++
+}
+
+async function _updateTemplateOrder(id, version, customFieldsRaw, stateKey) {
+  const schedule = customFieldsRaw.find(
+    (attr) => attr.name === 'schedule'
+  ).value
+  const cronExpression = parser.parseExpression(schedule)
+  const nextDeliveryCronDate = cronExpression.next()
+  const nextDeliveryDate = nextDeliveryCronDate.toDate()
+  const nextDeliveryDateISOString = nextDeliveryDate.toISOString()
+  const updateActions = [
+    {
+      action: 'setCustomField',
+      name: 'nextDeliveryDate',
+      value: nextDeliveryDateISOString,
+    },
+  ]
+
+  if (stateKey !== ACTIVE_STATE)
+    updateActions.push({
+      action: 'transitionState',
+      state: {
+        typeId: 'state',
+        key: ACTIVE_STATE,
+      },
+    })
+
+  const reminderDays = customFieldsRaw.find(
+    (attr) => attr.name === 'reminderDays'
+  )?.value
+  if (reminderDays) {
+    nextDeliveryCronDate.setDate(nextDeliveryCronDate.getDate() - reminderDays)
+    const nextReminderDateISOString = nextDeliveryCronDate.toISOString()
+    updateActions.push({
+      action: 'setCustomField',
+      name: 'nextReminderDate',
+      value: nextReminderDateISOString,
+    })
+  }
+
+  await updateOrderWithRetry(apiRoot, id, updateActions, version)
+}
+
+function _isError(response) {
+  return response.status >= 400
+}
+
+function _isErrorRecoverable(response) {
+  return (
+    response.status > 500 ||
+    response.status === 401 ||
+    response.status === 403 ||
+    response.status === 409
+  )
+}
+
+async function _fetchOrderBySubscriptionTemplateOrderRefAndDeliveryDate(
+  id,
+  deliveryDate
+) {
+  const query = `
+    query CheckExistingSubscriptionOrder($where: String) {
+      orders(where: $where, limit: 1) {
+        results {
+          version
+        }
+      }
+    }
+  `
+  const response = await ctpClient.queryGraphQl(query, {
+    where: `custom(fields(subscriptionTemplateOrderRef="${id}" AND deliveryDate="${deliveryDate}"))`,
+  })
+  return response.body?.data?.orders?.results?.[0]
+}
+
+function _buildQueryForTemplateOrdersThatNeedSubscriptionOrders(validStates) {
```

Comment:
```suggestion
function _buildQueryForTemplateOrdersToCreateSubscriptionOrders(stateIds) {
```

Reply:
So you did not like the function name change ?

Reply:
But I also feel this might stay as is :) so resolving it.

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/10/commits/1808de2d36edaac627bd22d4a14eb7992fc98a92